src/main/java/net/minecraftforge/registries/GameData.java

```diff
@@ -386,6 +392,10 @@ public static void revert(RegistryManager state, ResourceLocation registry, bool
                 revertTo(RegistryManager.VANILLA, false);
                 LOGGER.fatal("Detected errors during registry event dispatch, roll back to VANILLA complete");
             }
+            ModLoader.get().postEvent(new EntityAttributeCreationEvent());
```

Comment:
You know it is so incredibly typical of you guys that I literally predicted this exact comment the day I submitted this PR 34 days ago.

So let us review.

1. I introduced a new mod loading stage between registration and FMLCommonSetup event as that is the proper time to complete entity attribute operations.
2. Ya'll hate that solution and LITERALLY tell me to do it in the post stage callback for the registration stage. Read up a few posts and you will see that being suggested in here as the alternative. Do you know how hard it is to get anything done up to ya'lls standards when those standards are a shifting mass of bad suggestions in between the good with no indication as to which you believe to be which, and with almost no communication at all when questions are asked?

SO WHICH IS IT

1. It is ok to do it in the post stage callback which is currently "revert to vanilla"

2. Do it in a mod loading stage that you have adamantly insisted is not correct.

As I have mentioned previously in this very thread about this feature, it is ideal for us to have this done before FMLCommonSetup because this entire entity attribute setup bit is basically part of the registration phase. One of the whole points of having this entity attribute api is to that by the time we get to FMLCommonSetup we will have a full picture of what entity attributes look like.

WHERE ELSE WOULD YOU SUGGEST IT GO TO FULFILL THAT?

Reply:
You can pick anywhere at all in the vanilla code to throw a fucking hook.
The lifecycle shit is a mess, it's cpw's baby I don't like touching it, so ask him.

But Hell You could even add a fucking 'if we are not reverting to vanilla because shit is broken, then lets do things' and it would be better.
Right now you're "Hey I know there is an error and we LITERALLY just reverted everything to vanilla to try and gracefully handle it... But lets modify more shit!"
Look, Vanilla minecraft is complete shit in this regard. The ONLY sane way to do this is to re-write the entire god damn thing so that vanilla isn't a mess of static maps and functions.
But that isn't going to happen. So at least apply some common sense and logical thinking about the code you're working near.

Reply:
I did originally think about this and my original thought was so we're reverting our registries but does that mean that further modifications don't occur at all? All this revert does is return the registries to their original state, as far as I'm aware mods continue to execute otherwise. It seemed to me that doing it this way preserved the expected execution as much as possible.  

What if you wanted to add vanilla attributes to vanilla entities? That would still be entirely possible even after we revert the registries to vanilla, which is why I didn't make it conditional on not reverting originally. 

Obviously (now) it is your opinion that the opposite choice should be made and that this should not execute if we revert, and I'll get to putting together a solution that looks like that, but you must admit that the route I took is not fucking illogical it just is a different interpretation of what reverting to vanilla implies for this post-registration step of linking attributes to entity types. 

There's not really anywhere in "vanilla" to throw this fucking hook because it needs to be thrown after registration and before FMLCommonSetup. It needs to be part of the lifecycle shit. I didn't just come in here fucking willy-nilly not paying attention to shit. I read through (and understood even though I agree it is a mess) ya'lls lifecycle shit and tried to make sense of how it should be extended in this case. Just because ya'll dont like your own shit doesn't mean you need to take it out on me, a random guy just trying to contribute some sensible extensions to the lifecycle so that everyone doesn't have to rely on the much flakier AT and delayed execution approach that is necessary at the moment. Which by the way is what ya'll keep begging people to do instead of just coremod and AT shit to all hell.

Reply:
The point of the revert to vanilla is to obtain a known good state to continue falling through the error path.
It's the only sane way to handle an error deep in the code without hard crashing.

After this, the event busses are shut down and mods stop running.

Adding further modifications to this known-good state defeats the point, and might, if a mod were coded badly enough, cause the game to hard-crash anyway.

The best option is to else this if statement so that it runs if, and only if, the game is still in a valid state.

Reply:
To further that point, this could be literally one extra line and it'd be a practically suitable solution for this problem.
A single `else {`. That's all it needs.

Reply:
Thanks for being explicit about the intent of revert to vanilla curle. From just reading the lifecycle code it was not obvious what the final result should be as at this particular time in execution event busses are still active, mods are not unloaded, and theoretically it could still execute on only vanilla registry data. Surely ya'll can understand that from an outsider just reading the code and trying to piece together the various opinions dropped in this PR and associated issue perspective this was a valid decision to be made (albeit the unwanted one)? 

 Also I have acknowledged that I will be making this change. I'm not fighting for it not to be made. I am simply pointing out that I am not a fucking idiot and that it wasn't entirely clear which would be the better option.

The fundamental issue with this PR is that it involves doing something that currently does not have any equivalent. Mainly post registration modification of other global state. It is obvious that no one knew (or at least if they did know they did not bother to share) how they wanted this to look prior to me starting this PR and that feedback has been very flippant (and sometimes outright incorrect to the point of it being obvious that no one has bothered actually reading the relevant parts) with very little in the way of answering questions of clarification throughout the process. 

Look: I love forge, I love ya'll for developing and maintaining it. That's why I came here to try to contribute to an area of the code that was clearly being neglected. I would just encourage ya'll to read through this thread, make note of exactly how long the gaps are between responses (over 30 days on average, I originally opened this PR 4 months ago in October and literally multiple times have been given apologies of "sorry we should have gotten to this sooner/github bugged didnt see it"), exactly how many times a question just gets no response at all, exactly how many times there has been a given point of feedback that is literally entirely incorrect, and then exactly how many times needless aggression has been lobbied towards me for making decisions that have a non-obvious resolution. Also if it is not obvious enough, meeting ya'lls aggression with aggression has produced 100% better results than simply asking nicely and hoping someone will engage in good faith.